language/java/language_test.go

```diff
@@ -249,3 +249,119 @@ func TestLanguageExecute(t *testing.T) {
 		})
 	})
 }
+
+func TestMistakes(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Language *Language
+
+		RepositoryPath string
+
+		AssertMistakes func(repositoryPath string, mistakes []string)
```

Comment:
If the temporary path is a problem, you can go through the actual mistakes in the `validate` and `strings.TrimPrefix(actualMistakes[i], temporaryPath)`.

Reply:
The trim doesn't work well because AFAIK in Windows the temporary path will have back slashes but by looking into the CI logs the function returns the path with forward slashes, so the trim is not taking place.

Also, for some reason, on MacOs, the function is returning paths in the form of `/private/<temporary-path>`.

I did it in a slightly different way, but still using `ExpectedMistakes []string`

Reply:
so maven returns the errors on windows with file path forward slashes? but whyyyyyyy

Reply:
Here's the failed CI job where you can see the windows paths: https://github.com/symflower/eval-dev-quality/actions/runs/9466371364/job/26077902331